### PR TITLE
kubeadm: improve quickstart instructions

### PR DESCRIPTION
**Pull request by amacneil**
**What this PR does / why we need it**:

Improves instructional output following setup of a kubernetes master with kubeadm.

This helps prevent unnecessary support overhead such as: https://github.com/kubernetes/kubeadm/issues/212#issuecomment-292015402

**Example current output**:

```
To start using your cluster, you need to run (as a regular user):

  sudo cp /etc/kubernetes/admin.conf $HOME/
  sudo chown $(id -u):$(id -g) $HOME/admin.conf
  export KUBECONFIG=$HOME/admin.conf
```

**Example new output**:

```
To start using your cluster, you need to run (as a regular user):

  mkdir -p $HOME/.kube
  sudo cp -i /etc/kubernetes/admin.conf $HOME/.kube/config
  sudo chown $(id -u):$(id -g) $HOME/.kube/config
```

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by rmmh** [comment]
@k8s-bot gce etcd3 e2e test this

**Comment by amacneil** [comment]
Not clear why the e2e tests failed, so I rebased on latest master.

**Comment by amacneil** [comment]
@mikedanese @rmmh I think this requires someone to kick off the tests again.

**Comment by mikedanese** [comment]
@k8s-bot ok to test

**Comment by luxas** [comment]
Do we _really_ want to recommend the user to overwrite the default kubeconfig file?
The amount of commands to paste aren't reduced in the default case (just if you happen to switch shell)

The best solution would be something like: https://github.com/kubernetes/kubernetes/pull/38861, but we don't have that yet...

**Comment by wojtek-t** [comment]
@k8s-bot pull-kubernetes-unit test this

**Comment by mikedanese** [comment]
@k8s-bot pull-kubernetes-unit test this

**Comment by mikedanese** [comment]
@luxas why add do not merge

**Comment by amacneil** [comment]
@luxas I think it is worth making this the default, to improve the experience for users new to kubernetes. Any advanced/experienced users who already have another cluster up and running will know enough to merge the file (or copy it to a different location). However, new users are almost certainly going to open a new shell soon after installing their cluster, and will immediately be confused about how to run commands against it (as evidenced in the comment I linked to above).

If you are worried about users accidentally overwriting some existing config, as a failsafe we could use `cp -i` in the instructions so that overwriting an existing file will prompt the user. Let me know if you think it is worth updating the PR with this.

**Comment by luxas** [comment]
I wanted a justification/response to https://github.com/kubernetes/kubernetes/pull/44125#issuecomment-304699808

Thanks @amacneil for that comment! I do think the `-i` option makes a lot of sense, please update to use that.

kubeadm UX topic
/cc @jbeda @timothysc @lukemarsden

**Comment by amacneil** [comment]
Thanks @luxas. Updated with `cp -i` and rebased.

**Comment by luxas** [comment]
Thanks @amacneil! LGTM to me.

I want to have @jbeda's and @lukemarsden's eyes on this one as well though before removing do-not-merge...

**Review by luxas — APPROVED** [review]

**Review by gtaylor — APPROVED** [review]

**Comment by timothysc** [comment]
@k8s-bot pull-kubernetes-e2e-kops-aws test this 

@k8s-bot pull-kubernetes-unit test this

**Comment by gtaylor** [comment]
Why is this k8s-ci-robot laughing? This is unsettling.

**Pull request merged**
